addons/stock/models/res_partner.py

```diff
@@ -20,7 +20,5 @@ class ResPartner(models.Model):
 
     def action_view_stock_serial(self):
         action = self.env["ir.actions.act_window"]._for_xml_id("stock.action_production_lot_form")
-        action.update({
-            'domain': [('partner_ids', '=', self.id)],
-        })
+        action['domain'] = ['|', ('partner_ids', 'in', self.id), ('partner_ids.commercial_partner_id', 'in', self.id)]
```

Comment:
```suggestion
        action['domain'] = [('partner_ids', 'child_of', self.ids)]
```
Suppose the following partners :
- Company
  - Individual 1 (related to Company)
      - Extra address

If you send a serial to each partner, if you click on the serial/lots from Individual 1, you wouldn't see the lots sent in Extra address, as for both of them their `commercial_partner_id` would be Company.

Reply:
Indeed, I agree, it's better like this.
I was misled by @dalaOdoo 
Thanks for the review.